Source/Initialization/WarpXInitData.cpp

```diff
@@ -859,6 +885,34 @@ WarpX::InitLevelData (int lev, Real /*time*/)
        }
     }
 
+    // Reading external fields from data file
+    if (add_external_B_field && lev == 0) {
+        std::string read_fields_from_path="./";
+        pp_warpx.query("read_fields_from_path", read_fields_from_path);
+#if defined(WARPX_DIM_RZ)
+        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][0].get(), "B", "r");
+        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][1].get(), "B", "t");
+        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][2].get(), "B", "z");
```

Comment:
Feels like it ought to be this:

```diff
> git diff
diff --git a/Source/Initialization/WarpXInitData.cpp b/Source/Initialization/WarpXInitData.cpp
index af8f7c0e..36a4ec7e 100644
--- a/Source/Initialization/WarpXInitData.cpp
+++ b/Source/Initialization/WarpXInitData.cpp
@@ -891,8 +891,8 @@ WarpX::InitLevelData (int lev, Real /*time*/)
         pp_warpx.query("read_fields_from_path", read_fields_from_path);
 #if defined(WARPX_DIM_RZ)
         ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][0].get(), "B", "r");
-        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][1].get(), "B", "t");
-        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][2].get(), "B", "z");
+        ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][1].get(), "B", "z");
+        //ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][???].get(), "B", "t");
 #else
         ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][0].get(), "B", "x");
         ReadExternalFieldsFromFile(read_fields_from_path, Bfield_fp_external[lev][1].get(), "B", "y");
@@ -904,8 +904,8 @@ WarpX::InitLevelData (int lev, Real /*time*/)
         pp_warpx.query("read_fields_from_path", read_fields_from_path);
 #if defined(WARPX_DIM_RZ)
         ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][0].get(), "E", "r");
-        ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][1].get(), "E", "t");
-        ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][2].get(), "E", "z");
+        ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][1].get(), "E", "z");
+        //ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][???].get(), "E", "t");
 #else
         ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][0].get(), "E", "x");
         ReadExternalFieldsFromFile(read_fields_from_path, Efield_fp_external[lev][1].get(), "E", "y");
```

But that doesn't seem to produce sensible resulting fields

Reply:
Basically, whatever we do here needs to end up consistent with `AddExternalFields` just above